### ISSUE TITLE
STAT -l broken since 4.7.7

### ISSUE THREAD

**Issue by mpolden** [lavv17/lftp#365]
Listing directories over the control connection (e.g. `STAT -l`) seems to be broken in 4.7.7+.

Always empty reply on 4.7.7 and 4.8.0:
```
user@host:/$ quote STAT -al
---> STAT -al
213- status of -al:
```

4.7.6 lists directory as expected:
```
user@host:/$ quote STAT -al
---> STAT -al
213- status of -al:
<dir1>
<dir2>
...
213 End of Status
```

This basically means that with `ftp:use-stat-for-list` set to `true`, `ls` is broken.

**Comment by mpolden** [comment]
Example using ftp.no.debian.org:

4.8.0:
```
$ debug 5
$ rels
---> STAT .
<--- 213-Status follows:
<--- 213 End of status
```

Turning up the debug level shows that the expected data is received, but not parsed for some reason:

```
$ debug 10
$ rels
0:0 translated to pair 0:0 (0,0)
0 translated to pair 0:0 (0,0)
---> STAT .
<--- 213-Status follows:
<--- drwxr-xr-x    8 ftp      ftp            17 Jun 02 16:43 .
<--- drwxr-xr-x    8 ftp      ftp            17 Jun 02 16:43 ..
<--- -rw-r--r--    1 ftp      ftp          1216 Jun 02 16:42 HEADER.html
<--- lrwxrwxrwx    1 ftp      ftp             3 Mar 16  2010 Public -> pub
<--- drwxr-xr-x    3 ftp      ftp            10 Jun 30 08:04 about
<--- drwxrwxr-x   21 ftp      ftp            23 Jul 25 17:59 cdimage
<--- drwxr-xr-x    2 ftp      ftp             3 Jun 14  2006 conspiracy
<--- lrwxrwxrwx    1 ftp      ftp            22 Mar 16  2010 debian -> cdimage/.debian-mirror
<--- drwxr-xr-x    5 ftp      ftp             7 Mar 13  2016 debian-backports
<--- lrwxrwxrwx    1 ftp      ftp            16 Mar 16  2010 debian-cd -> cdimage/release/
<--- -rw-r--r--    1 ftp      ftp           318 Mar 21  2005 favicon.ico
<--- drwxr-xr-x   72 ftp      ftp            79 Jul 15 22:39 mirror
<--- drwxr-xr-x    4 ftp      ftp            12 Feb 05  2016 pub
<--- lrwxrwxrwx    1 ftp      ftp            23 Mar 16  2010 releases -> mirror/ubuntu-releases/
<--- -rw-r--r--    1 ftp      ftp          1859 Mar 23 13:53 robots.txt
<--- lrwxrwxrwx    1 ftp      ftp            12 Aug 01  2016 tails -> mirror/tails
<--- lrwxrwxrwx    1 ftp      ftp            13 Mar 16  2010 ubuntu -> mirror/ubuntu
<--- 213 End of status
copy: get hit eof
copy: waiting for put confirmation
copy: put confirmed store
copy: get is finished - all done
```

4.7.6 does the right thing:
```
$ rels
0:0 translated to pair 0:0 (0,0)
0 translated to pair 0:0 (0,0)
---> STAT .
<--- 213-Status follows:
<--- drwxr-xr-x    8 ftp      ftp            17 Jun 02 16:43 .
<--- drwxr-xr-x    8 ftp      ftp            17 Jun 02 16:43 ..
<--- -rw-r--r--    1 ftp      ftp          1216 Jun 02 16:42 HEADER.html
<--- lrwxrwxrwx    1 ftp      ftp             3 Mar 16  2010 Public -> pub
<--- drwxr-xr-x    3 ftp      ftp            10 Jun 30 08:04 about
<--- drwxrwxr-x   21 ftp      ftp            23 Jul 25 17:59 cdimage
<--- drwxr-xr-x    2 ftp      ftp             3 Jun 14  2006 conspiracy
<--- lrwxrwxrwx    1 ftp      ftp            22 Mar 16  2010 debian -> cdimage/.debian-mirror
<--- drwxr-xr-x    5 ftp      ftp             7 Mar 13  2016 debian-backports
<--- lrwxrwxrwx    1 ftp      ftp            16 Mar 16  2010 debian-cd -> cdimage/release/
<--- -rw-r--r--    1 ftp      ftp           318 Mar 21  2005 favicon.ico
<--- drwxr-xr-x   72 ftp      ftp            79 Jul 15 22:39 mirror
<--- drwxr-xr-x    4 ftp      ftp            12 Feb 05  2016 pub
<--- lrwxrwxrwx    1 ftp      ftp            23 Mar 16  2010 releases -> mirror/ubuntu-releases/
<--- -rw-r--r--    1 ftp      ftp          1859 Mar 23 13:53 robots.txt
<--- lrwxrwxrwx    1 ftp      ftp            12 Aug 01  2016 tails -> mirror/tails
<--- lrwxrwxrwx    1 ftp      ftp            13 Mar 16  2010 ubuntu -> mirror/ubuntu
<--- 213 End of status
drwxr-xr-x    8 ftp      ftp            17 Jun 02 16:43 .
copy: get hit eof
copy: waiting for put confirmation
drwxr-xr-x    8 ftp      ftp            17 Jun 02 16:43 ..
-rw-r--r--    1 ftp      ftp          1216 Jun 02 16:42 HEADER.html
lrwxrwxrwx    1 ftp      ftp             3 Mar 16  2010 Public -> pub
drwxr-xr-x    3 ftp      ftp            10 Jun 30 08:04 about
drwxrwxr-x   21 ftp      ftp            23 Jul 25 17:59 cdimage
drwxr-xr-x    2 ftp      ftp             3 Jun 14  2006 conspiracy
lrwxrwxrwx    1 ftp      ftp            22 Mar 16  2010 debian -> cdimage/.debian-mirror
drwxr-xr-x    5 ftp      ftp             7 Mar 13  2016 debian-backports
lrwxrwxrwx    1 ftp      ftp            16 Mar 16  2010 debian-cd -> cdimage/release/
-rw-r--r--    1 ftp      ftp           318 Mar 21  2005 favicon.ico
drwxr-xr-x   72 ftp      ftp            79 Jul 15 22:39 mirror
drwxr-xr-x    4 ftp      ftp            12 Feb 05  2016 pub
lrwxrwxrwx    1 ftp      ftp            23 Mar 16  2010 releases -> mirror/ubuntu-releases/
-rw-r--r--    1 ftp      ftp          1859 Mar 23 13:53 robots.txt
lrwxrwxrwx    1 ftp      ftp            12 Aug 01  2016 tails -> mirror/tails
lrwxrwxrwx    1 ftp      ftp            13 Mar 16  2010 ubuntu -> mirror/ubuntu
copy: put confirmed store
copy: get is finished - all done
```

**Comment by lavv17** [comment]
should be fixed by 30b26b5dc5e1b3ce0d0f6eedd9cdc09be880320f
